tests/ballerina-spec-conformance-tests/src/test/resources/5-values-types-and-variables/int.balt

```diff
@@ -0,0 +1,60 @@
+//  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+//  WSO2 Inc. licenses this file to you under the Apache License,
+//  Version 2.0 (the "License"); you may not use this file except
+//  in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+Test-case: succeed
+Label: int
+Author: SandaruJayawardana
+
+public function main() {
+    int x = 404;
+    io:println(x); // @output 404
+
+    int y = 0x1fe;
+    io:println(y); // @output 510
+
+    y = 0x1Fd;
+    io:println(y); // @output 509
+
+    y = 0X1fB;
+    io:println(y); // @output 507
+
+    x = y;
+    io:println(x); // @output 507
+
+    x = -9223372036854775808;
+    io:println(x); // @output -9223372036854775808
+
+    x = 9223372036854775807;
+    io:println(x); // @output 9223372036854775807
+}
+
+Test-case: error
+Label: int
+Author: SandaruJayawardana
+
+public function main() {
+    int x = -9223372036854775809;
+    io:println(x); // @error <Integer -9223372036854775809' too small>
+}
+
+Test-case: error
+Label: int
+Author: SandaruJayawardana
+
+public function main() {
+    int x = 9223372036854775808;
+    io:println(x); // @error <Integer '9223372036854775808' too large>
+}
```

Comment:
There's potentially few more tests like
- `DecimalNumber` cannot start with `0` unless the value itself is `0`. e.g., `01` should result in an error.
- `HexIntLiteral` must have both the `HexIndicator` and `HexNumber`

But these could be parser tests too. Let's discuss during the review and decide whether or not we should be adding such tests here.